import-scripts/annotation_suite/standardize_mutation_data.py

```diff
@@ -0,0 +1,1414 @@
+        if ref_allele[0] == alt_allele[0] and ref_allele != alt_allele:
+            # shift ref allele and alt allele by one nucleotide, set as "-" if len == 1
+            if len(ref_allele) == 1:
+                ref_allele = "-"
+            else:
+                ref_allele = ref_allele[1:]
+            if len(alt_allele) == 1:
+                alt_allele = "-"
+            else:
+                alt_allele = alt_allele[1:]
+
+            # fix start position value
+            if start_pos != "":
+                start_pos = str(int(start_pos) + 1)
+
+        # resolve variant type, end position, and variant class
+        variant_type = resolve_vcf_variant_type(ref_allele, alt_allele)
+        end_pos = resolve_end_position(vcf_data, start_pos, variant_type, ref_allele)
+        variant_class = resolve_variant_classification(vcf_data, variant_type, ref_allele, alt_allele)
+
+    maf_data["Variant_Classification"] = variant_class
+    maf_data["Variant_Type"] = variant_type
+    maf_data["Reference_Allele"] = ref_allele
+    maf_data["Tumor_Seq_Allele1"] = ref_allele
+    maf_data["Tumor_Seq_Allele2"] = alt_allele
+    maf_data["Start_Position"] = start_pos
+    maf_data["End_Position"] = end_pos
+    return maf_data
+
+def create_maf_record_from_vcf(sample_id, center_name, sequence_source, vcf_data, is_germline_data, matched_normal_sample_id, tumor_sample_data_col):
+    """
+        Creates MAF record from VCF data.
+    """
+    # init maf record
+    maf_data = init_maf_record()
+
+    # set easy to resolve values
+    maf_data["Tumor_Sample_Barcode"] = sample_id
+    maf_data["Matched_Norm_Sample_Barcode"] = matched_normal_sample_id
+    maf_data["Center"] = center_name
+    maf_data["Hugo_Symbol"] = resolve_hugo_symbol(vcf_data)
+    maf_data["Chromosome"] = resolve_chromosome(vcf_data)
+    maf_data["Sequence_Source"] = resolve_sequence_source(vcf_data, sequence_source)
+    maf_data["Verification_Status"] = DEFAULT_VERIFICATION_STATUS
+    maf_data["FILTER"] = vcf_data.get("FILTER", "")
+    if is_germline_data:
+        maf_data["Mutation_Status"] = "GERMLINE"
+    else:
+        maf_data["Mutation_Status"] = DEFAULT_MUTATION_STATUS
+
+
+    # resolve counts and variant allele data
+    resolve_vcf_counts_data(vcf_data, maf_data, matched_normal_sample_id, tumor_sample_data_col)
+    resolve_vcf_variant_allele_data(vcf_data, maf_data)
+    resolve_vcf_matched_normal_allele_data(vcf_data, maf_data, matched_normal_sample_id)
+    return maf_data
+
+def is_valid_vcf_maf_record(maf_record):
+    return not ((maf_record["Reference_Allele"] == "N" or maf_record["Reference_Allele"] == "") and maf_record["Tumor_Seq_Allele2"] == "")
+
+def get_vcf_file_header(filename):
+    vcf_file_header = []
+    with open(filename, "r") as vcf_file:
+        for line in vcf_file.readlines():
+            if line.startswith("#CHROM"):
+                vcf_file_header = list(map(str.strip, line.replace("#", "").split("\t")))
```

Comment:
for safety, I would change to line.replace("#", "", 1) so that only the first number sign is removed. If the header has additional number signs (such as an illegal trailing comment) we should let that generate errors rather than possibly interpreting a trailing comment as a sample id. Maybe we should do:
vcf_file_header_line = line.replace("#", "", 1)
if "#" in vcf_file_header_line:
   error message and exit
vcf_file_header = list(map(str.strip, vcf_file_header_line.split("\t"))

Reply:
the VCF file has a fixed header with a little flexibility in the tumor column name - I don't think this will be necessary after all